import_export/resources.py

```diff
@@ -694,7 +694,7 @@ def import_row(self, row, instance_loader, using_transactions=True, dry_run=Fals
                     # validate_instance(), where they can be combined with model
                     # instance validation errors if necessary
                     import_validation_errors = e.update_error_dict(import_validation_errors)
-                if self.skip_row(instance, original):
+                if not import_validation_errors and self.skip_row(instance, original):
```

Comment:
So the only concern I have with this approach is if skip_row is being overridden by people's code, now validate_instance will be called if there are any validation errors (where previously that might be intentionally overridden). 

Previously we trusted skip_row to make that determination. 

Is it possible skip_row could be fixed to correctly return False, without knowledge of the validation errors?

Reply:
Thanks for reviewing.  You make a very valid point.

To put it another way, there is now a risk that `skip_row()` won't be called.  I see that this could break code that has been around for a while.

I had a look at the PRs related to this code ([here](https://github.com/django-import-export/django-import-export/commit/517f10199f125f3c0387ac1d0417a9a554e17e10) and [here](https://github.com/django-import-export/django-import-export/pull/852/files)) to see if this it was deliberate choice to ignore validation errors for `skip_row()`.  I don't think it was a deliberate choice and I think it is a valid bug.

However, since we are changing `skip_row()` in release 3, so we could pass `validation_errors` in and the implementer can decide whether they want to handle the errors or not.  For example:

```
def skip_row(self, instance, original, row, validation_errors=None):
  if validation_errors:
    return False
```

This means that if there are validation errors, then they will be reported correctly.  This is probably the safest approach (and anyone who wants to ignore validation errors can do so).

Looping in @manelclos for his opinion.

Reply:
I agree with that approach in v3. The more we can allow people to control behavior in an intelligent way the better.

Reply:
If you think this fix is valuable enough to push through for 2.7x I'm fine with it too, I doubt it's likely that many people are using skip row in this way.

Reply:
I've updated this to be a v3 change, so some modifications.  Would be grateful for your review.